### PR TITLE
Added parser to read through vanilla_stats 

### PR DESCRIPTION
**Pull request by bornaehsani**
And generate a execution_stats.log file that prints out total execution time, number and fraction of each instruction executed, number and fraction of stalls and different types of misses.

### CODE REVIEW

File: software/py/generate_stats.py [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+    self.print_stats_timing()
+    self.print_stats_instructions()
+    self.print_stats_stalls()
+    self.print_stats_miss()
+
+
+
+
+  # default stats generator
+  def generate_stats(self, input_file):
```
Comment (tommydcjung) [inline software/py/generate_stats.py]: I think there is a csv module for python.
https://docs.python.org/3.6/library/csv.html

File: software/py/generate_stats.py [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+
+stalls_list = ['stall_fp_remote_load','stall_fp_local_load',\
+               'stall_depend','stall_depend_remote_load',\
+               'stall_depend_local_load','stall_force_wb',\
+               'stall_ifetch_wait','stall_icache_store',\
+               'stall_lr_aq','stall_md,stall_remote_req','stall_local_flw']
+
+
+
+class Stats:
```
Comment (tommydcjung) [inline software/py/generate_stats.py]: I would call it something more descriptive like VanillaStatsParser.
the filename should match the name of the python class in the file.

File: software/py/generate_stats.py [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+#   ex) python3 generate_stats.py 4 4 vanilla_stats.log
+#
+#   {manycore_dim_y}  Mesh Y dimension of manycore
+#   {manycore_dim_x}  Mesh X dimension of manycore
+
+
+
+import sys
+import os
+import re
```
Comment (tommydcjung) [inline software/py/generate_stats.py]: 're' not used?

**Comment by bornaehsani** [comment]
Stale. Replaced by PR #181 .

**Pull request closed without merging**